lib/puppet/type/archive.rb

```diff
@@ -150,6 +150,16 @@ def should_to_s(value)
     desc 'extract command group (using this option will configure the archive file permisison to 0644 so the user can read the file).'
   end
 
+  newparam(:proxy_type) do
+    desc 'proxy type (none|ftp|http|https)'
+    newvalues(:none, :ftp, :http, :https)
+    defaultto(:none)
+  end
+
+  newparam(:proxy_server) do
+    desc 'proxy address to use when accessing source'
+  end
+
```

Comment:
@nanliu But is this correct?

`proxy_type` is used to set an environment variable such as `http_proxy` or `https_proxy`.  It should default to the scheme of `source` not the scheme of `proxy_server`?